framework/CodeInterfaces/PHISICS/PhisicsRelapInterface.py

```diff
@@ -0,0 +1,257 @@
+    """
+    keyWordDict = {}
+    count = 0
+    for inFile in currentInputFiles:
+      keyWordDict[inFile.getType().lower()] = count
+      count = count + 1
+    return keyWordDict
+
+  def finalizeCodeOutput(self,command,output,workingDir):
+    """
+      This method is called by the RAVEN code at the end of each run (if the method is present, since it is optional).
+      It can be used for those codes, that do not create CSV files to convert the whatever output format into a csv
+      This methods also calls the method 'mergeOutput' if MPI mode is used, in order to merge all the output files into one.
+      @ In, command, string, the command used to run the just ended job
+      @ In, output, string, the Output name root
+      @ In, workingDir, string, current working dir
+      @ Out, None
+    """
+    # RELAP
+    self.Relap5Interface.finalizeCodeOutput(command,self.outFileName,workingDir)
+    # PHISICS
+    self.PhisicsInterface.finalizeCodeOutput(command,output,workingDir,phiRel=True)
+    import combine
+    jobTitle = self.PhisicsInterface.jobTitle
+    self.perturbationNumber = workingDir.split('/')[-1]
+    combine.combine(workingDir,os.path.join(workingDir,self.outFileName+'.csv'),os.path.join(workingDir,jobTitle+'-'+self.perturbationNumber+'.csv'),self.depTimeDict,self.inpTimeDict)
+
+  def checkForOutputFailure(self,output,workingDir):
+    """
+      This method is called by the RAVEN code at the end of each run  if the return code is == 0.
+      This method needs to be implemented by the codes that, if the run fails, return a return code that is 0
+      This can happen in those codes that record the failure of the job (e.g. not converged, etc.) as normal termination (returncode == 0)
+      This method can be used, for example, to parse the outputfile looking for a special keyword that testifies that a particular job got failed
+      (e.g. in RELAP5 would be the keyword "********")
+      @ In, output, string, the Output name root
+      @ In, workingDir, string, current working dir
+      @ Out, None
+    """
+    self.Relap5Interface.checkForOutputFailure(output,workingDir)
+  
+  def tailorRelap5InputFiles(self,currentInputFiles):
+    """
+      Generates a list of relap5 files only, in order to pass it to the RELAP parsers.
+      @ In, currentInputFiles, list,  list of current input files (input files from last this method call)
+      @ Out, relap5CurrentInputFiles, list, list of RELAP current input files (input files from last this method call)
+      @ Out, phisicsCurrentInputFiles, list, list of PHISICS current input files (input files from last this method call)
+    """
+    self.defineRelapInput()
+    relap5CurrentInputFiles = []
+    phisicsCurrentInputFiles = []
+    for index,inFile in enumerate(currentInputFiles):
+      if inFile.getType() in set(self.relapInputTypes):
```

Comment:
I thought we agreed we have to hardcode the types?

Reply:
I do not agree in hardcoding the types that are not needed (such as "h20", "rellic", etc.). The interface should not even know what a "rellic" is, because it does not need it